Sources/Commands/PackageCommands/ShowExecutables.swift

```diff
@@ -0,0 +1,88 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2014-2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import ArgumentParser
+import Basics
+import CoreCommands
+import Foundation
+import Workspace
+
+struct ShowExecutables: AsyncSwiftCommand {
+    static let configuration = CommandConfiguration(
+        abstract: "List the available executables from this package.")
+
+    @OptionGroup(visibility: .hidden)
+    var globalOptions: GlobalOptions
+
+    @Option(help: "Set the output format.")
+    var format: ShowExecutablesMode = .flatlist
```

Comment:
This should be made consistent with `ShowDependenciesMode` in [`SwiftPackageCommand`](https://github.com/swiftlang/swift-package-manager/blob/6ad5efb4445c213820897a8784461bc9a7f1314b/Sources/Commands/PackageCommands/ShowDependencies.swift#L72). The help string should be in a similar format, and if possible should have the same options (I believe it supports text, dot (??), json and flatlist)